solr/core/src/java/org/apache/solr/handler/admin/api/SetNestedClusterProperty.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.handler.admin.api;
+
+import static org.apache.solr.security.PermissionNameProvider.Name.COLL_EDIT_PERM;
+
+import jakarta.inject.Inject;
+import org.apache.solr.client.api.endpoint.SetNestedClusterPropertyApi;
+import org.apache.solr.client.api.model.SetNestedClusterPropertyRequestBody;
+import org.apache.solr.client.api.model.SolrJerseyResponse;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.jersey.PermissionName;
+
+public class SetNestedClusterProperty extends ClusterPropertiesAPIBase
+    implements SetNestedClusterPropertyApi {
+
+  @Inject
+  public SetNestedClusterProperty(CoreContainer coreContainer) {
+    super(coreContainer.getZkController().getZkClient());
+  }
+
+  @Override
+  @PermissionName(COLL_EDIT_PERM)
+  public SolrJerseyResponse createOrUpdateNestedClusterProperty(
+      SetNestedClusterPropertyRequestBody requestBody) {
+    SolrJerseyResponse response = instantiateJerseyResponse(SolrJerseyResponse.class);
+    try {
+      clusterProperties.setClusterProperties(requestBody.properties);
+    } catch (Exception e) {
+      throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "Error in API", e);
```

Comment:
[0] Not the most helpful error message, but I can see it came straight from the pre-existing "set-obj-property" API.

Commenting here as a reminder to myself to fix this and a few other places in a subsequent PR.